### PR TITLE
Backport Bitcoin PR#8715: net: only delete CConnman if it's been created

### PR DESCRIPTION
**Pull request by OlegGirko**
This is backport of Bitcoin PR bitcoin/bitcoin#8715.

The original PR description follows.
---
This fixes a possible shutdown crash. Thanks to @morcos for reporting.

In the case of (for example) an already-running bitcoind, the shutdown sequence begins before CConnman has been created, leading to a null-pointer dereference when g_connman->Stop() is called.

### CODE REVIEW

File: src/init.cpp [hunk 1]
```diff
@@ -228,7 +228,6 @@ void PrepareShutdown()
 #endif
     GenerateBitcoins(false, 0, Params(), *g_connman);
```
Comment (tgflynn) [inline src/init.cpp]: If g_connman has not yet been created here won't this dereference also cause a crash ?

Comment (OlegGirko) [inline src/init.cpp]: If `fGenerate` (first argument of `GenerateBitcoins()`) is `false`, `connman` argument is not used.

Comment (tgflynn) [inline src/init.cpp]: The dereference operator is still being called though in the function call even if the function doesn't make use of it.

This can result in undefined behavior if the stored pointer is null : http://www.cplusplus.com/reference/memory/unique_ptr/operator*/

> The unique_ptr shall not be empty (i.e., its stored pointer shall not be a null pointer) in order to be dereferenciable. 

I suggest this be fixed in another PR.
